sdk/python/feast/permissions/client/auth_client_manager_factory.py

```diff
@@ -15,8 +19,15 @@
 
 def get_auth_client_manager(auth_config: AuthConfig) -> AuthenticationClientManager:
     if auth_config.type == AuthType.OIDC.value:
-        assert isinstance(auth_config, OidcClientAuthConfig)
-        return OidcAuthClientManager(auth_config)
+        intra_communication_base64 = os.getenv("INTRA_COMMUNICATION_BASE64")
+        # If intra server communication call
+        if intra_communication_base64:
```

Comment:
pls track a GH issue to refactor this code and introduce a `IntraCommunicationClientManager` instead

Reply:
FYI @tmihalac @lokeshrangineni